app/logic/events.py

```diff
@@ -501,3 +502,77 @@ def getEventRsvpCount(eventId):
+    - If an event happens tomorrow but less than 24 hours away from us we still say that it happens 
+    tomorrow with no mention of the hour. 
+    - If an event happens tomorrow but more than 24 hours away from us, we'll count the number of days 
+    and hours in actual time.
+
+    E.x. if the current time of day is greater than the event start's time of day, we give a number of days 
+    relative to this morning and exclude all hours and minutes
+
+    On the other hand, if the current time of day is less or equal to the event's start of day we can produce 
+    the real difference in days and hours without the aforementioned simplifying language.
+    """
+
+    if currentDatetime is None:
+        currentDatetime = datetime.datetime.now().replace(second=0, microsecond=0)  
+    currentMorning = currentDatetime.replace(hour=0, minute=0)
+
+    eventStart = datetime.datetime.combine(event.startDate, event.timeStart)
+    eventEnd = datetime.datetime.combine(event.endDate, event.timeEnd)
+    
+    if eventEnd < currentDatetime:
+        return "Already passed"
+    elif eventStart <= currentDatetime <= eventEnd:
+        return "Happening now"
+    
+    timeUntilEvent = relativedelta(eventStart, currentDatetime)
+    calendarDelta = relativedelta(eventStart, currentMorning)
+    calendarYearsUntilEvent = calendarDelta.years
+    calendarMonthsUntilEvent = calendarDelta.months
+    calendarDaysUntilEvent = calendarDelta.days
+
+    yearString = f"{calendarYearsUntilEvent} year{'s' if calendarYearsUntilEvent > 1 else ''}"
+    monthString = f"{calendarMonthsUntilEvent} month{'s' if calendarMonthsUntilEvent > 1 else ''}"
+    dayString = f"{calendarDaysUntilEvent} day{'s' if calendarDaysUntilEvent > 1 else ''}"
+    hourString = f"{timeUntilEvent.hours} hour{'s' if timeUntilEvent.hours > 1 else ''}"
+    minuteString = f"{timeUntilEvent.minutes} minute{'s' if timeUntilEvent.minutes > 1 else ''}"
+    
+    # Years until
+    if calendarYearsUntilEvent: 
+        if calendarMonthsUntilEvent:
+            return f"{yearString} and {monthString}"
+        return f"{yearString}"
+    # Months until
+    if calendarMonthsUntilEvent:
+        if calendarDaysUntilEvent:
+            return f"{monthString} and {dayString}"
+        return f"{monthString}"
+    # Days until
+    if calendarDaysUntilEvent:
+        if eventStart.time() < currentDatetime.time():
+            if calendarDaysUntilEvent == 1:
+                return "Tomorrow"
+            return f"{dayString}"
+        if timeUntilEvent.hours:
+            return f"{dayString} and {hourString}"
+        return f"{dayString}"
+    # Hours until
+    if timeUntilEvent.hours:
+        if timeUntilEvent.minutes:
+            return f"{hourString} and {minuteString}"
+        return f"{hourString}"
+    # Minutes until
+    elif timeUntilEvent.minutes:
+        return f"{minuteString}"
+    # Seconds until
+    return "<1 minute"
```

Comment:
The case with <1 minute is not working. I was trying to set an event that is happening in less than a minute and I only got "(1 minute)" that was displayed.

Reply:
Thanks! Good catch!